b3lb/rest/celery/records.py

```diff
@@ -14,5 +14,51 @@
 # You should have received a copy of the GNU Affero General Public License
 # along with this program. If not, see <https://www.gnu.org/licenses/>.
 
-def celery_record():
-    pass
+import os
+import subprocess as sp
+from django.core.files.base import ContentFile
+from rest.models import Record, RecordProfile, RecordSet
+from django.conf import settings as st
+
+
+def check_dir_paths(sub_folder_name):
```

Comment:
We shouldn't use a hard-coded path which could contain orphan files from other tenants. We should use [`tempfile.TemporaryDirectory`](https://docs.python.org/3/library/tempfile.html#tempfile.TemporaryDirectory) with *ignore_cleanup_errors* wrapped in a context.

This would also allow to run multiple instances on the same working directory.

Reply:
Used `[TemporaryDirectory](https://docs.python.org/3/library/tempfile.html#tempfile.TemporaryDirectory) instead of own implementation.

But `ignore_cleanup_errors` is only available in python 3.10+, but pypy is running with python 3.7.

Reply:
IMHO `ignore_cleanup_errors` is no hard requirement, could be later added if pypy 3.10 is available (pypy 3.9 is already available)